StandardLibrary/src/StandardLibrary.dfy

```diff
@@ -335,7 +335,7 @@ module StandardLibrary {
    * The function is compilable, but will not exhibit enviable performance.
    */
 
-  function method {:tailrecursion} SetToOrderedSequence<T(!new,==)>(s: set<seq<T>>, less: (T, T) -> bool): (q: seq<seq<T>>)
+  function SetToOrderedSequence<T(!new,==)>(s: set<seq<T>>, less: (T, T) -> bool): (q: seq<seq<T>>)
```

Comment:
This will be breaking for consumers of the MPL, since you're changing it from compiled to ghost, but there are compiled references to it already, e.g. https://github.com/aws/aws-database-encryption-sdk-dynamodb/blob/main/DynamoDbEncryption/dafny/DynamoDbEncryption/src/DynamoToStruct.dfy#L1091

Are you relying on assuming we can replace all such references with `ComputeSetToSequence[2]`?

Reply:
Ah, I removed internal compiled references to it but didn't realize it was part of the exported API.

I suspect we should be able to replace them all, though it's possible there are complexities I'm not thinking of.

Reply:
I think the two instances in that file are the only place it's used in a compiled context, and I think it can be replaced by the `Compute` version there.

Reply:
While we can logically replace them, we cannot assume that the code on our customers computer will not be updated.

i.e: Our customers will still have references to the old version, and rely on the references being provided by the MPL.

Unless both of the references in the DB-ESDK, and any references in the ESDK, are only in ghost code and have ALWAYS been in only in ghost code.

Reply:
@atomb Could you make this a `function by method` instead and use the `ComputeSetToSequence2` implementation in the by method body?

Reply:
Actually I don't think you have to change this anyway, I don't get an error when I do `dafny verify src/StandardLibrary.dfy --function-syntax:3 --enforce-determinism`.

Since this `:|` is in an expression context it already has to be provably deterministic.

Reply:
Oh, you are indeed correct. I've changed it back.